.travis.yml

```diff
@@ -1,44 +1,176 @@
-env:
-   - CABALVER=1.18 GHCVER=7.4.1 # Debian 7
-   - CABALVER=1.20 GHCVER=7.6.3 # Debian 8
-   - CABALVER=1.18 GHCVER=7.8.4
-   - CABALVER=1.22 GHCVER=7.10.3
-   - CABALVER=1.24 GHCVER=8.0.2
-   - CABALVER=1.24 GHCVER=8.2.2
-   - CABALVER=2.2 GHCVER=8.4.3
-   - CABALVER=2.4 GHCVER=8.6.3
-   - CABALVER=head GHCVER=head
-
+# This Travis job script has been generated by a script via
+#
+#   haskell-ci '--ghc-head' 'Glob.cabal'
+#
+# For more information, see https://github.com/haskell-CI/haskell-ci
+#
+# version: 0.3.20190429
+#
+language: c
+dist: xenial
+git:
+  # whether to recursively clone submodules
+  submodules: false
+cache:
+  directories:
+    - $HOME/.cabal/packages
+    - $HOME/.cabal/store
+before_cache:
+  - rm -fv $CABALHOME/packages/hackage.haskell.org/build-reports.log
+  # remove files that are regenerated by 'cabal update'
+  - rm -fv $CABALHOME/packages/hackage.haskell.org/00-index.*
+  - rm -fv $CABALHOME/packages/hackage.haskell.org/*.json
+  - rm -fv $CABALHOME/packages/hackage.haskell.org/01-index.cache
+  - rm -fv $CABALHOME/packages/hackage.haskell.org/01-index.tar
+  - rm -fv $CABALHOME/packages/hackage.haskell.org/01-index.tar.idx
+  - rm -rfv $CABALHOME/packages/head.hackage
 matrix:
-   allow_failures:
-      - env: GHCVER=head CABALVER=head
-
+  include:
+    - compiler: ghc-8.8.1
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-8.8.1","cabal-install-3.0"]}}
+      env: GHCHEAD=true
+    - compiler: ghc-8.6.5
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-8.6.5","cabal-install-2.4"]}}
+    - compiler: ghc-8.4.4
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-8.4.4","cabal-install-2.4"]}}
+    - compiler: ghc-8.2.2
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-8.2.2","cabal-install-2.4"]}}
+    - compiler: ghc-8.0.2
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-8.0.2","cabal-install-2.4"]}}
+    - compiler: ghc-7.10.3
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-7.10.3","cabal-install-2.4"]}}
+    - compiler: ghc-7.8.4
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-7.8.4","cabal-install-2.4"]}}
+    - compiler: ghc-7.6.3
+      addons: {"apt":{"sources":["hvr-ghc"],"packages":["ghc-7.6.3","cabal-install-2.4"]}}
```

Comment:
I sympathise with that, but unfortunately `haskell-ci` won't support older `cabal-install`s. Supporting `v1-build` and `v2-build`s in one script is just too tricky.

I fixed few `-Werror` woes, feel free to cherry-pick them if you decide to reject this PR otherwise.